File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:
    a = []
    def __init__(self, nums: List[int]):
        self.a =nums

    def sumRange(self, left: int, right: int) -> int:
        return sum(self.a[left:right+1])        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.js
/**
 * @param {number[]} nums
 */
var NumArray = function(nums) {
 this.nums = nums;
}
    ;

/** 
 * @param {number} left 
 * @param {number} right
 * @return {number}
 */
NumArray.prototype.sumRange = function(left, right) {
    let sum=0
    for(let i=left;i<=right;i++){
        sum+=this.nums[i]
    }
    return  sum

};

/** 
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */